telemetry_package/main.py

```diff
@@ -3,164 +3,203 @@
+            self.server_sock = socket.socket(
+                socket.AF_UNIX, socket.SOCK_STREAM)
             self.server_sock.bind(self.chosen_socket_path)
+            print(
+                f"Listening for connection on the UNIX-socket at {self.chosen_socket_path} ...")
             self.server_sock.listen(1)
 
             # Accept the connection
             self.client_sock, _ = self.server_sock.accept()
-            print('Connection accepted. Creating subscription')
-
+            print('Connection accepted. Creating subscriptions...')
 
+            # Subscribe to /rosout
             self.rosout_subscription = self.create_subscription(
                 Log,
                 '/rosout',
-                self.__rosout_callback, 
-                qos_profile = qos_profile)
-            
+                self.__rosout_callback,
+                qos_profile=qos_profile)
+
             self.rosout_subscription
             print('Created subscription to "/rosout".')
 
-
-            self.control_subscription = self.create_subscription(
-                Control,
-                TopicNames.Control,
-                qos_profile = qos_profile)
-
-            self.control_subscription 
-            print('Created subscription to "control".')
-
-
+            # Subscribe to TopicNames.Heartbeat
             self.heartbeat_subscription = self.create_subscription(
                 Heartbeat,
-                TopicNames.Heartbeat, 
+                TopicNames.Heartbeat,
                 self.__heartbeat_callback,
-                qos_profile = qos_profile)
-            
+                qos_profile=qos_profile)
+
             self.heartbeat_subscription
             print('created subscription to "heartbeat".')
 
-            # Deactivating the node as according to the docs, could result in an error? (unsure yet)
-            self._deactivate_()
-
         except Exception as e:
             print(f'Error occurred: {e}')
             self.server_sock.close()
             os.remove(self.chosen_socket_path)
-        
 
-    def __rosout_callback(self, log_msg)->None:
-        '''
-        Callback function for the /rosout subscription. 
+    def throttle_log_message(self, log_msg) -> bool:
+        """
+        Using self.THROTTLE_LOG_MESSAGE_DEFINITION and self._message_throttle_table, this function return a bool based on, if a message should be send or not with regard to the minimum time to pass between to subsequent messages
+        defined in self.THROTTLE_LOG_MESSAGE_DEFINITION.
+
+        Args:
+            log_msg: the incoming log message from the /rosout topic
+
+        Returns:
+            True, if message should be throttled (not send). False, if the message can be send.
+
+        """
+        if log_msg.name in self.THROTTLE_LOG_MESSAGE_DEFINITION:
+            if log_msg.name in self._message_throttle_table and (time.time() - self._message_throttle_table[log_msg.name])*1000 < self.THROTTLE_LOG_MESSAGE_DEFINITION[log_msg.name]:

```

Comment:
```suggestion
            if log_msg.name in self._message_throttle_table and (time.time() - self._message_throttle_table[log_msg.name])*1000 > self.THROTTLE_LOG_MESSAGE_DEFINITION[log_msg.name]:

```

Reply:
Alright, but then we add this to line 152.
Have to ask @J0las then, if there is anything on Warning, Error or Fatal.
Also I am hoping, that the sub-loggers get implemented, then throttling can be done more precise.